File: test-suite-kotlin2-ksp2/src/test/kotlin/io/micronaut/docs/http/client/bind/method/NameAuthorization.kt
package io.micronaut.docs.http.client.bind.method;

import io.micronaut.context.annotation.AliasFor
import io.micronaut.core.bind.annotation.Bindable
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.FUNCTION

//tag::clazz[]
@MustBeDocumented
@Retention(RUNTIME)
@Target(FUNCTION) // <1>
@Bindable
annotation class NameAuthorization(val name: String = "")

//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/http/client/bind/method/NameAuthorization.py
from typing import Annotated

from micronaut.context.annotation import AliasFor
from micronaut.core.bind.annotation import Bindable


# tag::clazz[]
@Bindable
def NameAuthorization(
    value: Annotated[str, AliasFor(member="name")] = "",
    name: Annotated[str, AliasFor(member="value")] = "",
):
    def decorator(func):
        return func
    return decorator
# end::clazz[]
